src/main/java/org/accounting/system/endpoints/MetricDefinitionEndpoint.java

```diff
@@ -908,4 +919,63 @@ public Response getAllAccessControl(){
 
         return Response.ok().entity(response).build();
     }
+
+    @Tag(name = "Metric Definition")
+    @Operation(
+            operationId = "search-metric-definition",
+    summary = "Searches a new Metric Definition.",
+    description = "Searches a metric definition ")
+    @APIResponse(
+            responseCode = "201",
+            description = "Metric Definition has been filtered successfully.",
+            content = @Content(schema = @Schema(
+                    type = SchemaType.ARRAY,
+                    implementation = MetricDefinitionResponseDto.class)))
+
+     @APIResponse(
+            responseCode = "401",
+            description = "User/Service has not been authenticated.",
+            content = @Content(schema = @Schema(
+                    type = SchemaType.OBJECT,
+                    implementation = InformativeResponse.class)))
+
+    @APIResponse(
+            responseCode = "500",
+            description = "Internal Server Errors.",
+            content = @Content(schema = @Schema(
+                    type = SchemaType.OBJECT,
+                    implementation = InformativeResponse.class)))
+    @SecurityRequirement(name = "Authentication")
+    @APIResponse(
+            responseCode = "403",
+            description = "The authenticated user/service is not permitted to perform the requested operation.",
+            content = @Content(schema = @Schema(
+                    type = SchemaType.OBJECT,
+                    implementation = InformativeResponse.class)))
+    @APIResponse(
+            responseCode = "404",
+            description = "Metric Definition has not been found.",
+            content = @Content(schema = @Schema(
+                    type = SchemaType.OBJECT,
+                    implementation = InformativeResponse.class)))
+    @APIResponse(
+            responseCode = "500",
+            description = "Internal Server Errors.",
+            content = @Content(schema = @Schema(
+                    type = SchemaType.OBJECT,
+                    implementation = InformativeResponse.class)))
+    @SecurityRequirement(name = "Authentication")
+
+    @POST
+    @Path("/search")
+
+    @Produces(value = MediaType.APPLICATION_JSON)
+    @Consumes(value = MediaType.APPLICATION_JSON)
+    @Permission(operation = org.accounting.system.enums.Operation.READ, collection = Collection.MetricDefinition)
+
+    public Response search(@Valid @NotNull(message = "The request body is empty.") @RequestBody String json, @Context UriInfo uriInfo) throws  NoSuchFieldException, ParseException {
```

Comment:
It may be helpful to add a .txt file which contains a request with a json filter